### PR TITLE
concat: init

### PR DESCRIPTION
**Pull request by 4z3**
###### Motivation for this change

The `concat` builder was introduced to allow creating wrapper derivations that retain access to other resources, like manpages.    

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by grahamc** [comment]
is this different from buildEnv?

File: pkgs/build-support/trivial-builders.nix [hunk 1]
```diff
@@ -167,4 +167,30 @@ rec {
...
+  #         exec ${super.hello}/bin/hello "$@"
+  #       '')
+  #     ];
+  #   };
+  # }
+  #
+  concat = name: xs: runCommand name {} ''
+    mkdir $out
+    ${lib.flip lib.concatMapStrings xs (x: ''
+      cp --remove-destination -vrs ${x}/* $out
```
Comment (Mic92) [inline pkgs/build-support/trivial-builders.nix]: How about adding `--link` to cp to save space/time?

File: pkgs/build-support/trivial-builders.nix [hunk 2]
```diff
@@ -167,4 +167,30 @@ rec {
...
 
+
+  # Combine a list of derivations using symlinks.  Paths in later derivations
+  # take precedence over earlier ones.
+  #
+  # Example: create wrapper but retain all other files (man pages etc.)
+  #
+  # {
+  #   nixpkgs.config.packageOverrides = super: {
+  #     hello = pkgs.concat "hello" [
```
Comment (Mic92) [inline pkgs/build-support/trivial-builders.nix]: As this function is added to a rather big namespace, a more distinctive name is useful.
How about `pkgs.concatPackages`?

**Comment by Mic92** [comment]
The functionality provided by this function overlaps with `symlinkJoin`.

File: pkgs/build-support/trivial-builders.nix (hunk 2 above)
Comment (vcunat) [inline pkgs/build-support/trivial-builders.nix]: Yes `lib.concat` already exists (!)

**Pull request closed without merging**